packages/eslint-plugin-lit-a11y/lib/utils/HasLitHtmlImportRuleExtension.js

```diff
@@ -0,0 +1,161 @@
+  if (!contextSpecifiers.has(context)) {
+    const USER_LIT_HTML_SPECIFIERS = Array.isArray(context.settings.litHtmlSources)
+      ? context.settings.litHtmlSources
+      : [];
+    contextSpecifiers.set(context, [...DEFAULT_LIT_HTML_SPECIFIERS, ...USER_LIT_HTML_SPECIFIERS]);
+  }
+
+  return contextSpecifiers.get(context);
+}
+
+/**
+ * Given an import id like `lit-html/lit-html.js`, returns the package name i.e. `lit-html`
+ * @param {string|number|boolean|RegExp} id
+ * @return {string}
+ */
+function getPackageName(id) {
+  const moduleSpecifier = id.toString();
+  if (!moduleSpecifier.startsWith('@')) return id.toString().split('/')[0];
+
+  const [scope, pkg] = moduleSpecifier.split('/');
+  return [scope, pkg].join('/');
+}
+
+/**
+ * Is this ImportDeclaration importing lit-html, taking the user's settings into account
+ * @param {import('estree').ImportDeclaration & import('eslint').Rule.NodeParentExtension} node
+ * @param {LitA11yRuleContext} context
+ */
+function isLitHtmlImportDeclaration(node, context) {
+  /** @param {typeof node.specifiers[number]} specifier */
+  const isLitHtmlImport = specifier =>
+    (specifier.type && specifier.type === 'ImportNamespaceSpecifier') ||
+    (specifier.type === 'ImportSpecifier' &&
+      getLitHtmlSpecifiers(context).includes(getPackageName(node.source.value)));
+
+  const { specifiers = [] } = node;
+
+  return specifiers.some(isLitHtmlImport);
+}
+
+/**
+ * @param {import('estree').ImportSpecifier | import('estree').ImportDefaultSpecifier | import('estree').ImportNamespaceSpecifier} specifier
+ * @return {specifier is import('estree').ImportNamespaceSpecifier}
+ */
+function isNamespaceImport(specifier) {
+  return specifier.type === 'ImportNamespaceSpecifier';
+}
+
+/**
+ * Returns a list of lit-html tag function names
+ * @param {import('estree').ImportDeclaration & import('eslint').Rule.NodeParentExtension} node
+ * @return {string[]}
+ */
+function getLitHtmlTags(node) {
+  return node.specifiers
+    .map(specifier => {
+      switch (specifier.type) {
+        case 'ImportNamespaceSpecifier':
+          return 'html';
+        case 'ImportSpecifier':
+          if (specifier.imported.name === 'html') return specifier.local.name || 'html';
+          return '';
+        default:
+          return '';
+      }
+    })
+    .filter(Boolean);
+}
+
+/**
+ * Returns a list of lit-html tag function names
+ * @param {import('estree').ImportDeclaration & import('eslint').Rule.NodeParentExtension} node
+ * @return {string[]}
+ */
+function getLitHtmlNamespaces(node) {
+  return node.specifiers
+    .map(specifier => isNamespaceImport(specifier) && specifier.local.name)
+    .filter(Boolean);
+}
+
+const HasLitHtmlImportRuleExtension = {
+  /**
+   * @param {LitA11yRuleContext} context
+   * @return {import('eslint').Rule.RuleListener}
+   */
+  createAdditionalVisitors(context) {
+    /** if user set `litHtmlSources` to literal `false`, analyse all html-tagged literals */
```

Comment:
What is the usecase for setting `litHtmlSources` to `false`?

Reply:
Disable checking. I forgot to document that in readme